src/main/java/goodgenerator/blocks/tileEntity/FuelRefineFactory.java

```diff
@@ -217,8 +217,13 @@ protected CheckRecipeResult validateRecipe(@NotNull GT_Recipe recipe) {
             @Override
             protected GT_OverclockCalculator createOverclockCalculator(@NotNull GT_Recipe recipe,
                     @NotNull GT_ParallelHelper helper) {
-                return super.createOverclockCalculator(recipe, helper)
-                        .setSpeedBoost(1F / (float) (1 << (Tier - recipe.mSpecialValue)));
+                long consumedPower = (long) recipe.mEUt * (1L << (Tier - recipe.mSpecialValue));
+                return new GT_OverclockCalculator().setRecipeEUt(consumedPower)
+                        .setDuration(
+                                (int) Math.ceil(
+                                        (recipe.mDuration * helper.getDurationMultiplierDouble())
+                                                / (double) (1 << (Tier - recipe.mSpecialValue))))
+                        .setEUt(consumedPower);
```

Comment:
How about adding
```java
public static GT_OverclockCalculator ofNoOverclock(long eut, int duration) {
    return new GT_OverclockCalculator().setRecipeEUt(eut).setDuration(duration).setEUt(eut);
}
```
to GT? It would improve readability a bit

Reply:
https://github.com/GTNewHorizons/GT5-Unofficial/pull/2140